Hide message text of blocked contacts

cascheberg opened the pull request:
### Contributor checklist

<!-- mark with x between the brackets -->
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
  - HTC Sensation XE
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have made the choice whether I want the Bithub reward or not by omitting or adding the word `FREEBIE` in my commit message

---
### Description

Fixes #4237
- hide message text, delivery status and thumbnail

**before:**
![blocked_before](https://cloud.githubusercontent.com/assets/3845150/13976987/4568a514-f0c5-11e5-91cc-503632608e50.png)

**after:**
![blocked_after](https://cloud.githubusercontent.com/assets/3845150/13976991/54fcf318-f0c5-11e5-84bf-4b8d17af689e.png)

// FREEBIE

cascheberg commented:
It would also be possible to add two new special message types and insert those into the database when blocking / unblocking the user.

moxie0 commented on src/org/thoughtcrime/securesms/database/model/ThreadRecord.java:
```diff
@@ -65,6 +64,10 @@ public ThreadRecord(@NonNull Context context, @NonNull Body body, @Nullable Uri
 
+  public static SpannableString getBlockedDisplayBody(Context context) {
+    return emphasisAdded(context.getString(R.string.ThreadRecord_blocked));
+  }
```
i can't think of any reason to put this method here like this. if it's a decision the model needs to make, it should make it. but this is telling the model what decision to make.

moxie0 commented on src/org/thoughtcrime/securesms/database/model/ThreadRecord.java:
```diff
@@ -105,11 +108,11 @@ public SpannableString getDisplayBody() {
   }
 
-  private SpannableString emphasisAdded(String sequence) {
+  private static SpannableString emphasisAdded(String sequence) {
```
why

moxie0 commented on src/org/thoughtcrime/securesms/database/model/ThreadRecord.java:
```diff
@@ -105,11 +108,11 @@ public SpannableString getDisplayBody() {
   }
 
-  private SpannableString emphasisAdded(String sequence, int start, int end) {
+  private static SpannableString emphasisAdded(String sequence, int start, int end) {
```
why

moxie0 commented on src/org/thoughtcrime/securesms/ConversationListItem.java:
```diff
@@ -235,10 +236,20 @@ public void run() {
+      this.subjectView.setText(ThreadRecord.getBlockedDisplayBody(getContext()));
+      setThumbnailSnippet(null, null);
+    }
+  }
```
this is pretty side-effecty

cascheberg replied:
So that means that inserting special user blocked / unblocked message types really is the better solution?

cascheberg commented:
PR updated with review comments addressed.

cascheberg commented:
This PR also fixes an issue regarding asynchronous loading of recipient data that is displayed in a threads subject view.

cascheberg commented:
Moved some of the changes into new PR #5785, so that this PR is now really small.

moxie0 commented on src/org/thoughtcrime/securesms/ConversationListItem.java:
```diff
@@ -235,6 +235,11 @@ public void run() {
+
+        if (recipients.isBlocked()) {
+          setStatusIcons(null);
+          setThumbnailSnippet(null, null);
```
i don't like the idea of intentionally passing null into here, ideally those arguments should be annotated "@ NonNull"

cascheberg replied:
PR #5785 pulls the thread reference into the class, that would be a first step. Can you have a look at it?

cascheberg commented:
PR rebased, refactored and tested.
Fixed issue of passing `null` by introducing new `clearThumbnailSnippet` and `clearStatusIcons` methods, which leads to several whitespace changes.

cascheberg commented:
Closing this PR because I am not planning to follow up on it.
